lib/smart-commit-tokenizer.js

```diff
@@ -10,6 +10,7 @@ module.exports = function () {
   const newline = {match: /\n/, lineBreaks: true, next: 'main'}
   const time = {match: /(?<= )#time(?= )/, push: 'workLog'}
   const whitespace = /[ \t]+/
+  const notWhitespace = /[^\r\n\t ]+/
```

Comment:
Is there any reason we can't use `\s` to match any whitespace?

Reply:
Yes...sort of. There's some history here.

1. The `whitespace` pattern above must exclude newline and carriage return characters because they are used to delimit commands and must be matched separately. In order to accomplish this, I listed whitespace characters explicitly. My list didn't include unicode characters.
2. The `notWhitespace` pattern needed to match any characters that `whitespace` didn't. This meant it needed to match Unicode whitespace characters, which led to the pattern you see above. More details on this here: https://github.com/integrations/jira/pull/213/commits/c8ea4ecbb3ffac4210cb16436c62fc79cd670f9e

If characters didn't match at least one of those patterns, it was possible to trigger a syntax error, which we saw here.

Taking a fresh look at this code today, I realized a better, albeit brain-twisting, solution: match using a negated character set that includes non-whitespace (`\S`) and `\r\n`. This precisely excludes `\n\r` while including other unicode whitespace characters.

New code pushed!

Reply:
Ah this is great context. Thank you!
It might be nice to include this as a comment above this const so the context is inline 😄

Reply:
Already done! Do you think more detail is warranted? https://github.com/integrations/jira/pull/289/files#diff-742bcc471ec3bbc2c448c13a42e19feaR12

Reply:
nope! That's great. Thank you ❤️